validation/views.py

```diff
@@ -495,6 +502,20 @@ def user_is_linguist(user):
     return False
 
 
+def get_phrases_from_session(session, all_phrases):
+    phrases_from_session = []
+    session_date = datetime.datetime.strptime(session, "%Y-%m-%d").date()
+    for phrase in all_phrases:
+        for recording in phrase.recordings:
+            if (recording.session.date == session_date) and (
+                phrase not in phrases_from_session
```

Comment:
If there might be thousands of phrases, this could get slow, because checking if `phrase` is in the growing `phrases_from_session` list is O(n^2) whereas it could be only O(n) if you used a set.

Reply:
I was worried about this being very slow on production so I'll switch it over to a set.

Reply:
Now that I take a second look, there *should* be a way to get this out of the database directly with some sort of query…

Reply:
I was really hoping there would be, but I couldn't find a direct link between all three tables. And I don't think I can index across multiple tables

Reply:
It doesn’t match the name of the method if there can be multiple sessions per date, but finding all phrases from all sessions on a given date should be `Phrase.objects.filter(recording__session__date='2015-03-23')`

Reply:
This is great--so much nicer to look at! Thanks for figuring that out.

Reply:
:)